lib/spack/spack/build_environment.py

```diff
@@ -1149,7 +1149,7 @@ def child_fun():
     For more information on `multiprocessing` child process creation
     mechanisms, see https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
     """
-    parent_pipe, child_pipe = multiprocessing.Pipe()
+    parent_pipe, child_pipe = multiprocessing.Pipe(duplex=False)
```

Comment:
Since `duplex is False`, would you mind renaming these:
```suggestion
    read_pipe, write_pipe = multiprocessing.Pipe(duplex=False)
```
?